catt/controllers.py

```diff
@@ -104,91 +137,65 @@ def clear(self):
+        self.not_buffering = threading.Event()
 
-        if running_app == self._dmc_app_id:
-            self.dmc_ready.set()
-        elif running_app == self._yt_app_id:
-            self.yt_ready.set()
-
-        if state != "BUFFERING" and self.yt_ready.is_set():
-            self.queue_ready.set()
+        if app_id == active_app_id:
+            self.app_ready.set()
+            if state != "BUFFERING":
+                self.not_buffering.set()
 
     def new_cast_status(self, status):
-        if status.app_id == self._dmc_app_id:
-            self.dmc_ready.set()
-            self.yt_ready.clear()
-        elif status.app_id == self._yt_app_id:
-            self.yt_ready.set()
-            self.dmc_ready.clear()
+        if status.app_id == self.app_id:
+            self.app_ready.set()
         else:
-            self.dmc_ready.clear()
-            self.yt_ready.clear()
+            self.app_ready.clear()
 
     def new_media_status(self, status):
         if status.player_state == "BUFFERING":
-            self.queue_ready.clear()
-        elif self.yt_ready.is_set():
-            self.queue_ready.set()
+            self.not_buffering.clear()
+        elif self.app_ready.is_set():
+            self.not_buffering.set()
 
 
 class CastController:
-    def __init__(self, device_name, state_check=True):
-        cache = Cache()
-        cached_ip = cache.get(device_name)
-
-        try:
-            if not cached_ip:
-                raise ValueError
-            self.cast = pychromecast.Chromecast(cached_ip)
-        except (pychromecast.error.ChromecastConnectionError, ValueError):
-            self.cast = get_chromecast(device_name)
-            cache.set(self.cast.name, self.cast.host)
-
-        self.cast.wait()
-
-        self._listener = StatusListener(self.cast.app_id,
+    def __init__(self, cast, name, app_id, prep=None):
+        self.cast = cast
+        self.name = name
+        self._listener = StatusListener(app_id, self.cast.app_id,
                                         self.cast.media_controller.status.player_state)
         self.cast.register_status_listener(self._listener)
         self.cast.media_controller.register_status_listener(self._listener)
 
-        # We need to create the ytc object in the constructor
-        # as the cli is calling add_to_yt_queue multiple times
-        # when the user is casting a youtube playlist.
-        self._ytc = YouTubeController()
-        self.cast.register_handler(self._ytc)
-
-        if state_check:
-            self._check_state()
-
-    def _check_state(self):
-        if self.cast.app_id == "E8C28D3C" or not self.cast.app_id:
+        try:
+            self.cast.register_handler(self._controller)
+        except AttributeError:
+            self._controller = self.cast.media_controller
+
+        if prep == "app":
+            self._prep_app()
+        elif prep == "control":
+            self._prep_control()
+
+    def _prep_app(self):
```

Comment:
These methods probably need docstrings.

Reply:
Yep. I'll add them along with the one for ```setup_cast```.